src/web/handlers.rs

```diff
@@ -212,3 +195,46 @@ fn fallback_response(settings: &Settings, tiles: &cache::Tiles) -> HttpResponse
         tiles.to_response(settings.cache_control_header)
     }
 }
+
+/// Check if the tile request should be responded early.
+///
+/// This allows us to short circuit requests if:
+///   - they are not sent from regions of the live markets
+///   - they are sent from unknown device types (`form_factor == "other"`)
+///
+/// It returns a proper response if the early response is desired.
+/// Otherwise, it returns None.
+async fn maybe_early_respond(
+    settings: &Settings,
+    state: &web::Data<ServerState>,
+    location: &Location,
+    device_info: &DeviceInfo,
+) -> Option<HttpResponse> {
+    if !state
+        .partner_filter
+        .read()
+        .await
+        .all_include_regions
+        .contains(&location.country())
+    {
+        trace!("get_tiles: country not included: {:?}", location.country());
+        // Nothing to serve. We typically send a 204 for empty tiles but
+        // optionally send 200 to resolve
+        // https://github.com/mozilla-services/contile/issues/284
+        let response = if settings.excluded_countries_200 {
+            HttpResponse::Ok()
+                .content_type("application/json")
+                .body(EMPTY_TILES.as_str())
+        } else {
+            HttpResponse::NoContent().finish()
+        };
+        return Some(response);
+    }
+
+    if matches!(&device_info.form_factor, FormFactor::Other) {
```

Comment:
suggestion (non-blocking): Let's do this check first, before the live markets check above? That one looks more involved, and this is a simpler pattern match.